pkg/server/metrics.go

```diff
@@ -0,0 +1,96 @@
+package server
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/arquillian/ike-prow-plugins/pkg/log"
+	"github.com/arquillian/ike-prow-plugins/pkg/github/client"
+)
+
+// RegisterMetrics registers prometheus collectors to collect metrics
+func (s *Server) RegisterMetrics() []error {
+	errors := make([]error, 0, 3)
+	metrics := NewMetrics()
+	if rateLimitCollector, e := registerOrGet(metrics.RateLimit); e == nil {
+		metrics.RateLimit = rateLimitCollector.(*prometheus.GaugeVec)
+	} else {
+		errors = append(errors, e)
+	}
+
+	if webHookCollector, e := registerOrGet(metrics.WebHookCounter); e == nil {
+		metrics.WebHookCounter = webHookCollector.(*prometheus.CounterVec)
+	} else {
+		errors = append(errors, e)
+	}
+
+	if HandledEventsCollector, e := registerOrGet(metrics.HandledEventsCounter); e == nil {
+		metrics.HandledEventsCounter = HandledEventsCollector.(*prometheus.CounterVec)
+	} else {
+		errors = append(errors, e)
+	}
+
+	if len(errors) > 0 {
+		return errors
+	}
+
+	s.Metrics = metrics
+	return make([]error, 0);
```

Comment:
you should be able to return the original `errors` variable without checking its content - that is checked in the caller